### PR TITLE
Fix xmlrpc timeout, #4843

### PR DESCRIPTION
**Pull request by jim-parry**
Set timeout after socket open, for subsequent operations.
Eliminated usleep(...) in response handling loop
Updated user guide page for xmlrpc library

Signed-off-by:Master Yoda <jim_parry@bcit.ca>

### CODE REVIEW

**Comment by jim-parry** [comment]
@infinity0n3 Does this do the trick? There are no unit tests for xmlrpc, so could you test the change with your situation? Thanks

**Comment by infinity0n3** [comment]
I can confirm that it works. Thanks!

File: user_guide_src/source/libraries/xmlrpc.rst [hunk 1]
```diff
@@ -490,6 +490,10 @@ Class Reference
 
 			$this->xmlrpc->timeout(6);
 
+		This timeout period will be used both for an initial connection to 
+                the remote server, as well as for getting a response from it.
+                Make sure you set the timeout before calling `send_request`.
```
Comment (narfbg) [inline user_guide_src/source/libraries/xmlrpc.rst]: A single backqoute works in Markdown, but not in rst ... we need double backquotes here.

**Comment by narfbg** [comment]
Also, I haven't really looked into it ... is the `usleep()`, `continue` removal related to the timeout issue? No other side-effects?

File: user_guide_src/source/libraries/xmlrpc.rst (hunk 1 above)
Comment (jim-parry) [inline user_guide_src/source/libraries/xmlrpc.rst]: Noted, thanks. Should I edit that in place or provide a separate PR?

**Comment by jim-parry** [comment]
As near as I can tell, the usleep() was there to introduce an artificial delay waiting for the remote I/O to complete. Setting the stream timeout is a replacement for that.

File: user_guide_src/source/libraries/xmlrpc.rst (hunk 1 above)
Comment (narfbg) [inline user_guide_src/source/libraries/xmlrpc.rst]: Well, you can't edit a merged PR ... You could commit straight to develop though.

File: system/libraries/Xmlrpc.php [hunk 2]
```diff
@@ -734,6 +734,8 @@ public function sendPayload($msg)
 			.'User-Agent: '.$this->xmlrpcName.$r
 			.'Content-Length: '.strlen($msg->payload).$r.$r
 			.$msg->payload;
+		
+		stream_set_timeout($fp,$this->timeout); // set timeout for subsequent operations
```
Comment (narfbg) [inline system/libraries/Xmlrpc.php]: Also, there needs to be a space after the comma here.

**Comment by narfbg** [comment]
Did that myself with the commit above.

Also, referencing #4843 (apparently it being in the title doesn't work).

**Pull request merged**